tests/thumbnail.rs

```diff
@@ -33,16 +33,13 @@ fn thumbnail(
     };
 
     let cover_frame = loop {
-        let cover_packet = {
-            let mut cover_packet = None;
-            while let Some(packet) = input_format_context.read_packet()? {
+        let cover_packet = loop {
+            match input_format_context.read_packet()? {
                 // Get first video packet.
-                if packet.stream_index == video_stream_index as i32 {
-                    cover_packet = Some(packet);
-                    break;
-                }
+                Some(x) if x.stream_index == video_stream_index as i32 => break x,
+                Some(_) => {}
+                None => return Err("Can't find video cover packet".into()),
```

Comment:
```suggestion
                None => bail!("Can't find video cover packet"),
```
Using `bail` from `anyhow` would be better here.